### PR TITLE
Refactor register feature

### PR DESCRIPTION
**Pull request by rolandnsharp**
- modularize
- use async

TODO: write test

### CODE REVIEW

**Comment by dbkaplun** [comment]
Is this PR good for merge?  Looks like that "TODO: write test" is fixed, no?

**Comment by rolandnsharp** [comment]
@dbkaplun it's basically finished, but I'm still not happy with the tests. I'm only testing API calls. It's still very tedious to test every functionality path manually when reviewing PR's. If you have any ideas about how I'd better go about integration testing this thing I'd love to know. I'll merge this over the weekend regardless as I think it's okay.

File: test/register-test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        registerUser(user, function(err, res) {
+            should.not.exist(err);
+            user.username.should.be.exactly(res.username);
+            user.email.should.be.exactly(res.email);
+
+            done();
+        });
+    });
+
+    it.only('should error if username is taken', function(done) {
```
Comment (ThomWright) [inline test/register-test.js]: Should this be `only`? This means only this test will run. Also, it contains no assertion.

**Comment by ThomWright** [comment]
About the tests, I'm currently writing a bunch of API tests for the server. It's slow progress but it's a start.

I think additional testing here makes perfect sense, to ensure the CLI is using the APIs correctly.

File: test/register-test.js (hunk 1 above)
Comment (rolandnsharp) [inline test/register-test.js]: Yeah, I'm still working on the tests. I didn't mean to push them.

**Comment by rolandnsharp** [comment]
Thanks for the recomendation @ThomWright , I'll write tests for all the rest of the API calls as I refactor the rest of the code.

**Pull request merged**